lib/node/nodes/centroid.js

```diff
@@ -1,17 +1,32 @@
 'use strict';
 
 var Node = require('../node');
+var debug = require('../../util/debug')('analysis:centroid');
 
 var TYPE = 'centroid';
 
 var PARAMS = {
     source : Node.PARAM.NODE(Node.GEOMETRY.ANY),
     category_column : Node.PARAM.NULLABLE(Node.PARAM.STRING()),
-    aggregation: Node.PARAM.NULLABLE(Node.PARAM.STRING(), 'count'),
-    aggregation_column: Node.PARAM.NULLABLE(Node.PARAM.STRING())
+    aggregation: Node.PARAM.NULLABLE(Node.PARAM.ENUM('avg', 'count', 'max', 'min', 'sum'), null),
```

Comment:
If there is no category column the aggregation doesn't matter so you just don't do it. It will happen the same if there is no category column and the user provides an aggregation operation and an aggregation column, right?